pkg/rule/inmem.go

```diff
@@ -0,0 +1,118 @@
+package rule
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/lifesum/configsum/pkg/errors"
+)
+
+type inmemRuleState map[string]map[string]rule
+
+type inmemRepo struct {
+	rules inmemRuleState
+	ids   map[string]struct{}
+}
+
+// NewInmemRepo returns an in-memory backed Repo implementation.
+func NewInmemRepo() (Repo, error) {
+	return &inmemRepo{
+		rules: inmemRuleState{},
+		ids:   map[string]struct{}{},
+	}, nil
+}
+
+func (r *inmemRepo) Get(configID, name string) (rule, error) {
+	rules, ok := r.rules[configID]
+	if !ok {
+		return rule{}, errors.Wrap(errors.ErrNotFound, fmt.Sprintf("config id '%s'", configID))
+	}
+
+	rl, ok := rules[name]
+	if !ok {
+		return rule{}, errors.Wrap(errors.ErrNotFound, fmt.Sprintf("rule name '%s'", name))
+	}
+
+	return rl, nil
+}
+
+func (r *inmemRepo) Create(input rule) (rule, error) {
+	if _, ok := r.ids[input.id]; ok {
+		return rule{}, errors.Wrap(errors.ErrExists, "id")
+	}
+
+	if _, ok := r.rules[input.configID]; !ok {
+		r.rules[input.configID] = map[string]rule{}
+	}
+
+	if _, ok := r.rules[input.configID][input.name]; ok {
+		return rule{}, errors.Wrap(errors.ErrExists, "name")
+	}
+
+	r.rules[input.configID][input.name] = input
+
+	return r.rules[input.configID][input.name], nil
+}
+
+func (r *inmemRepo) UpdateWith(input rule) (rule, error) {
+	rl, err := r.Get(input.configID, input.name)
+	if err != nil {
+		return rule{}, err
+	}
+
+	rl.active = input.active
+	rl.activatedAt = input.activatedAt
+	rl.buckets = input.buckets
+	rl.configID = input.configID
+	rl.createdAt = input.createdAt
+	rl.criteria = input.criteria
+	rl.description = input.description
+	rl.endTime = input.endTime
+	rl.id = input.id
+	rl.kind = input.kind
+	rl.name = input.name
+	rl.startTime = input.startTime
+	rl.updatedAt = time.Now()
```

Comment:
Wouldn't it work as well to just overwrite the rule in the `inmemRuleState` with the input?